### PR TITLE
Export a function to add extensions to errors

### PR DESCRIPTION
**Pull request by lorenzo**
It is possible to add a path to errors, but it was not possible to add extensions. This exports a function to do so

### CODE REVIEW

File: morpheus-graphql-core/src/Data/Morpheus/Types/Internal/AST/Error.hs [hunk 1]
```diff
@@ -62,6 +63,10 @@ atPositions GQLError {..} pos = case toList pos of
   posList -> GQLError {locations = locations <> Just posList, ..}
 {-# INLINE atPositions #-}
 
+
+withExtensions :: GQLError -> Map Text Value -> GQLError
+withExtensions gqlerr ext = gqlerr { extensions = Just ext }
```
Comment (nalchevanidze) [inline morpheus-graphql-core/src/Data/Morpheus/Types/Internal/AST/Error.hs:68]: we will overwrite all existing extension with it. is it behavior that you intended?
what about this?
```hs
withExtensions err ext = err { extensions = extensions err  <> Just ext }
```

Comment (lorenzo) [reply morpheus-graphql-core/src/Data/Morpheus/Types/Internal/AST/Error.hs:68]: The behaviour is modelled after the `withPath` function. I'm happy with either solution

Comment (lorenzo) [reply morpheus-graphql-core/src/Data/Morpheus/Types/Internal/AST/Error.hs:68]: would you like me to change it to your suggestion? :)

Comment (nalchevanidze) [reply morpheus-graphql-core/src/Data/Morpheus/Types/Internal/AST/Error.hs:68]: yes. that would be right way to go. i think :)

**Comment by nalchevanidze** [comment]
hi, could you please separate this features in different pull requests?

**Comment by lorenzo** [comment]
@lorenzo sorry. Someone incorrectly pushed  to the wrong branch. I’ll clean up today

**Comment by lorenzo** [comment]
closing as this got messed up

**Pull request closed without merging**